### PR TITLE
Mergesort

### PR DESCRIPTION
**Pull request by xialiu1988**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: 401code_challenges/codeChallenge/src/main/java/mergeSort/App.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+  private static int[] merge(int[] arr,int left,int mid,int right){
+      int[] tmp = new int[arr.length];
+      int r1 = mid + 1;
+      int tIndex = left;
+      int cIndex=left;
+
+      while(left <=mid && r1 <= right) {
+          if (arr[left] <= arr[r1])
```
Comment (evanslaton) [inline 401code_challenges/codeChallenge/src/main/java/mergeSort/App.java]: Not using brackets on multi-line conditional statements is "dangerous!" It's very easy to cause little bugs that are hard to find
